packages/core/src/lib/implementation/execute-plugin.ts

```diff
@@ -123,15 +135,43 @@ export async function executePlugins(
 
     progressBar?.updateTitle(`Executing  ${chalk.bold(pluginCfg.title)}`);
 
-    const pluginReport = await executePlugin(pluginCfg);
-    progressBar?.incrementInSteps(plugins.length);
-
-    return outputs.concat(pluginReport);
-  }, Promise.resolve([] as PluginReport[]));
+    try {
+      const pluginReport = await executePlugin(pluginCfg);
+      progressBar?.incrementInSteps(plugins.length);
+      return outputs.concat({ value: pluginReport, status: 'fulfilled' });
+    } catch (e: unknown) {
+      progressBar?.incrementInSteps(plugins.length);
+      return outputs.concat({
+        error: e instanceof Error ? e.message : String(e),
+        status: 'rejected',
+      });
+    }
+  }, Promise.resolve([] as ExecutePluginResult[]));
 
   progressBar?.endProgress('Done running plugins');
 
-  return pluginsResult;
+  const errors = pluginsResult.filter(
```

Comment:
We have a helper method [`logMultipleResults`](https://github.com/code-pushup/cli/blob/b071ce9c444f74f81cb71cbd1c089801dbcc0531/packages/utils/src/lib/file-system.ts#L44). It is used here https://github.com/code-pushup/cli/blob/b071ce9c444f74f81cb71cbd1c089801dbcc0531/packages/core/src/lib/implementation/persist.ts#L87

Reply:
Moved and reused `logMultipleResults` logic for logging multiple results by `Promise.allSettled`. Reused both at `persist.ts` and `execute-plugin.ts`

Reply:
I would intro a helper function encapsulating the logging logic.

Reply:
Moved logging logic from `LogMultipleResults()` to `logPluginExecution()`, same file.